### PR TITLE
added rnd(num,precision) to maths.sc

### PR DESCRIPTION
**Pull request by Ghoulboy78**
This is the proper name in the Queens's English.
Also, this means that we can round more or less precisely based on how we want it, and it's gonna be a hell of a lot easier to do certain things with stuff like angles, where you want only a certain degree of accuracy

### CODE REVIEW

**Comment by Ghoulboy78** [comment]
made it a one-liner cos @gnembon is a genius and told me how to do all that complicated crap in one line

**Comment by Ghoulboy78** [comment]
But, there is also that fact that my convolutedly long method was actually taking the same time as the new one

File: programs/fundamentals/maths.sc [hunk 1]
```diff
@@ -40,4 +40,7 @@ distance_sq(vec1, vec2) -> reduce(vec1 - vec2, _a + _*_, 0);
 
 distance(vec1, vec2) -> sqrt(reduce(vec1 - vec2, _a + _*_, 0));
 
-dot(vec1, vec2) -> reduce(vec1 * vec2, _a * _, 0);
\ No newline at end of file
+dot(vec1, vec2) -> reduce(vec1 * vec2, _a * _, 0);
+
+//num is the number you're rounding, rnum is the number (10,100,0.1 but also 5) to which you are rounding
+rnd(num,precision)->(return(round(num/precision)*precision));
```
Comment (gnembon) [inline programs/fundamentals/maths.sc]:
```suggestion
rnd(num,precision)-> round(num/precision)*precision);
```

File: programs/fundamentals/math.sc [hunk 2]
```diff
@@ -40,4 +40,7 @@ distance_sq(vec1, vec2) -> reduce(vec1 - vec2, _a + _*_, 0);
 
 distance(vec1, vec2) -> sqrt(reduce(vec1 - vec2, _a + _*_, 0));
 
-dot(vec1, vec2) -> reduce(vec1 * vec2, _a * _, 0);
\ No newline at end of file
+dot(vec1, vec2) -> reduce(vec1 * vec2, _a * _, 0);
+
+//num is the number you're rounding, rnum is the number (10,100,0.1 but also 5) to which you are rounding
+rnd(num,precision)->(return(round(num/precision)*precision));
```
Comment (gnembon) [inline programs/fundamentals/math.sc]:
```suggestion
rnd(num,precision)-> round(num/precision)*precision;
```

**Comment by gnembon** [comment]
merge conflicts - will add that manually myself

**Pull request closed without merging**